tests/func/plots/test_collect.py

```diff
@@ -0,0 +1,59 @@
+import dpath
+
+from dvc.repo.plots import Plots
+
+
+def test_plots_definitions_works_with_nested_plots(tmp_dir, scm, dvc):
```

Comment:
This is good.

[C] A few things from this test for me:

1. breadcrumbs would be helpful, i.e. naming the variables to reflect what we are trying to test here. My understanding is that we are verifying that a plot definition from a sub-directory is not being overridden by its parent. 
2. This would be a good test to have in `tests/integration/plots/test_plots.py` and ensure that nothing untoward happens to the data later in the system. For example, does `step` get correctly assigned to `x` when no `x` is present?
3. If I remember correctly the `plot_id` can be a string like "acc vs loss". I don't think that information breaks the solution but I would like to see that tested as well.

Reply:
[Q] Does this work for `y` as well?

Reply:
Thanks for the review Matt :)
I tried to take your comments into account, let me know if we understood each others, or if I get you wrong.

Yes it works for `y` as well, I added a test condition to show you that ;)